### PR TITLE
WIP: feature: OpenAPI Specification

### PR DESCRIPTION
**Pull request by adamoxley**
Initial PR for OpenAPI Spec. Can be opened in something like https://stoplight.io, we can also export it as a .json file and it would then be importable to something like Postman.

I've put in some APIs that would likely suit the mobile app, but can tweak them as needed of course.

### CODE REVIEW

File: openapi/reference/presentations/models/presentation.v1.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+properties:
+  id:
+    type: string
+    format: uuid
+  title:
+    type: string
+    example: "What's new in Core Data"
+  synopsis:
+    type: string
+    example: "Nothing :("
```
Comment (SwiftyAlex) [inline openapi/reference/presentations/models/presentation.v1.yaml:17]: 😆

**Review by SwiftyAlex — COMMENTED** [review]
Just comments, looks great!

I think we need something for a `schedule` which sorta wraps around a couple of these? 

Ideally, It would give us an array of events, with timestamps, with attached speakers. Said events, or slots or something ( I suck at naming ) would have to support being attached to a talk, or something else like a lunch/coffee break

Attaching to a talk could just point straight to the presentation, but for the others it might be nice to just have title, duration, start/end, description (optional)

File: openapi/reference/speakers/models/speaker.basic.v1.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+xml:
+  name: Speaker Basic
+properties:
+  name:
+    type: string
+    example: Tim Cook
+  profile_image:
+    type: string
+    format: url
+  twitter:
```
[the remark this answers is not in the document]
Comment (SwiftyAlex) [reply openapi/reference/speakers/models/speaker.basic.v1.yaml:13]: idk why the text is all red and angry its not intentional sorry 😓

File: openapi/reference/speakers/speakers.api.v1.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  get:
+    summary: List all Speakers
+    operationId: listSpeakers
+    responses:
+      '200':
+        description: An array of Speakers
+        content:
+          application/json:
+            schema:
+              $ref: ./models/speaker.basic.v1.yaml
```
[the remark this answers is not in the document]
Comment (adamoxley) [reply openapi/reference/speakers/speakers.api.v1.yaml:11]: Added in a basic model at the OpenAPI spec side of things as /speakers returns speakers having presentations relationship on them. But we also have /presentations which has speakers... which have presentations on them 😂 bit of an issue.

Basic model is essentially _the_ model without the relationship it's being called from. Just how we'd model the responses really than it being a representation of the database model.

[the remark this answers is not in the document]
Comment (SwiftyAlex) [reply openapi/reference/speakers/speakers.api.v1.yaml:11]: Ahhhh ok, classic "includes" vibes

Makes total sense, nice way to handle it in the spec!

**Comment by Sherlouk** [comment]
Closing due to inactivity, since this PR was introduced newer solutions in the community exist for creating OpenAPI specs from your Vapor routes (and automatically generating a Stoplight documentation page). If desire still remains, I'd recommend looking in that direction to minimise the amount of code we need to maintain.

**Pull request closed without merging**